docs/misc/components.md

```diff
@@ -0,0 +1,105 @@
+Text Components
+==================
+
+`Component` is part of text that can be formatted, have styles applied, and be chained with other components.
+`MutableComponent` is currently the only implementation of `Component` whose style and list of chained components changed, it can be created by either passing its contents directly to `MutableComponent#create` or using one of these static helpers:
+
+* `Component#literal(String text)` - creates component whose represented text is simply `text` that was passed in.
+* `Component#nullToEmpty(@Nullable String text)` - same as `#literal` except it creates empty component if null has been passed 
+* `Component#translatable(String translationKey)` - creates component which is represented as localized text to user, read [internationalization] for more details.
+* `Component#empty()` - creates empty component
+* `Component#keybind(String name)` - creates component which is represented as name of a Keyboard key for passed keybind.
+* `Component#nbt(String path, boolean interpreting, Optional<Component> separator, DataSource dataSource)` - creates component for representing nbt data specified by `path` inside `dataSource`
+* `Component#score(String name, String objective)` - creates component for representing `objective`'s score of entity specified by [entity selector][selectors] `name`.
+* `Component#selector(String pattern, Optional<Component> separator)` - creates component for displaying names of entities selected by [entity selector][selectors] `pattern`.
+
+Applying Style
+==============
+
+`Style` specifies how component should look like, such as bold, italic, color, custom font, etc.
+Style is immutable objects and any modification to already existing style will instead create new instance. Its constructor is `private` so you'll have to copy and reconfigure `Style.EMPTY`.
+
+Style's default configurations are `null` instead of concrete value. This means not explicitly specified configuration may be overridden.
+After configuring style to your preference, you can apply it to your component with either `MutableComponent#setStyle` for overwriting previous style, or `#withStyle` for only applying non-null configurations to already existing style.
+
+Here's an example of how you can stylize your component:
+```java
+MutableComponent text = Component.literal("Hello!"); // Creates MutableComponent wrapping literal "Hello!"
+
+Style blueItalic = Style.EMPTY // Copies empty style and sets color to blue and makes it italic
+        .withColor(0x0000FF)
+        .withItalic(true);
+Style red = Style.EMPTY // Copies empty style and sets color to red
+        .withColor(0xFF0000);
+Style bold = Style.EMPTY // Copies empty style and makes it bold
+        .withBold(true);
+
+Style doubleLines = Style.EMPTY // Copies empty style and makes it both underlined and strikethrough
+        .withUnderlined(true)
+        .withStrikethrough(true);
```

Comment:
Put comments on top of code since otherwise people will have to scroll to read it.